### PR TITLE
Fix #15 clang tidy workflow bug

### LINKED ISSUE #15: [BUG] `clang-tidy` workflow fails
**Describe the bug**
I can't tell if this is an issue with the files or the workflow .yml but I noticed the clang-tidy github action is failing (on both issue #9  and #10 . I thought it was because we didn't have any C++ code but now we do. Needs further investigation.

**To Reproduce**
1. Make some changes
2. Submit pull request
3. See clang-tidy workflow has failed

**Expected behavior**
clang-tidy workflow passes

**Desktop (please complete the following information):**
 - OS: not specific - occurs in GH actions

**Additional context**

### PR DESCRIPTION
**Pull request by Bai-Li-NOAA**
This pull request addresses issue #15 
- I created a new [run-clang-tidy.yml](https://github.com/NOAA-FIMS/FIMS/blob/bug-issue-15-clang-tidy-workflow/.github/workflows/run-clang-tidy.yml) by modifying Google benchmark [clang-tidy workflow](https://github.com/google/benchmark/blob/main/.github/workflows/clang-tidy.yml)
- the new workflow is able to detect bad code in gtest/ .cpp files and their associated .hpp files
    - the workflow failed when I added bad code to gtest/ .cpp file or their associated .hpp files. the error messages could be found in this [log](https://github.com/NOAA-FIMS/FIMS/runs/5501703791?check_suite_focus=true)
    - I removed the bad code after testing the workflow and this pull request passed all checks. Yay!
 
Other changes:
- I removed Makefile from the main directory based on conversations during the modeling team meeting 
- I updated CMake files and tests based on the new folder structure under inst/include folder

Notes: 
The new clang-tidy checks won't check any .hpp files that are not used for building gtests. Lines 5- 110 of the workflow log shows the details of the clang-tidy checks. 
I changed the CMake required version number in the original [clang-tidy.yml](https://github.com/NOAA-FIMS/FIMS/blob/main/.github/workflows/clang-tidy.yml) file. It gives pass to bad code and I do not know what to do with this file.

### CODE REVIEW

File: .github/workflows/clang-tidy.yml
Comment (k-doering-NOAA) [inline .github/workflows/clang-tidy.yml:2]: is there a reason to keep this file? I thought this one doesn't work?

File: inst/include/population_dynamics/recruitment/recruitment.hpp
Comment (k-doering-NOAA) [inline inst/include/population_dynamics/recruitment/recruitment.hpp:34]: I'm assuming this code was added for testing purposes?

**Review by k-doering-NOAA — APPROVED** [review]
Thanks, Bai, for working on this! I see the test where the run-clang-tidy caught the errors (https://github.com/NOAA-FIMS/FIMS/runs/5501703791?check_suite_focus=true)

I'm actually not clear on what kind of checks clang does - does it do style checks (like will fail code if indents/ whitespace use is not correct) or just looks for compile errors? It might be nice to add a sentence or two to the top of the yml about what the workflow does since this is probably new to a lot of our team.

The only other suggestion I had was one about workflows when developing github action workflows - I see there are 12 commits in this PR, but I think you could squash them to get a more clean git history (of course, you had to push up to github to test github actions). What I would typically do if I'm working on a branch by myself is to use 'git rebase --interactive commitnum' to squash commits or delete ones that I dont need when I am done developing and ready to submit a PR. You could make a new branch before doing this or just use the same one (though you would force push up to github, which is only ok if you are confident no one is trying to modify your branch).

There also is a "squash commits" button when merging, which maybe does the same thing in a far easier way! I'm curious what you and @ChristineStawitz-NOAA  think about using this when developing github actions workflows in particular so that the messy commit history isn't preserved

Anyway, I approve this PR, as all the suggestions are fairly minor.

**Comment by k-doering-NOAA** [comment]
Also, I'll leave it to @ChristineStawitz-NOAA to decide when to merge in.

**Comment by Bai-Li-NOAA** [comment]
Thanks @k-doering-NOAA, for reviewing the code. Those are great suggestions.
- Clang-tidy does code diagnostics. I added more explanation to the workflow. To style C++ code, I suggest that we modify the [clang-format-lint workflow](https://github.com/google/benchmark/blob/main/.github/workflows/clang-format-lint.yml) from Google benchmark repo.
- I also squashed a few commits as you suggested. I think Git squash is a good practice. I usually use the "squash commits" button when merging a pull request if the commits could be squashed. It takes some time to practice, but it is important to have a clean commit history. I suggest that we encourage developers to squash their code locally before making a pull request. At the same time, for developers who do not feel comfortable to squash their commits right now, they can add a sentence in the pull request and say if they wish the reviewers to squash and merge their code.
- I think we can remove the old clang-tidy workflow. @ChristineStawitz-NOAA , let me know if you have any suggestions.

**Comment by k-doering-NOAA** [comment]
Looks great, Bai! Thanks for making changes so quickly- the explanation of clang was really helpful. I think this is ready to merge when you and Christine are ready.

**Pull request merged**
